fda/registration.py

```diff
@@ -4,38 +4,60 @@
 functional data and related routines, in basis form as well in discretized form.
 
 """
+import enum
 
 import numpy
 import scipy.integrate
 
 
-def _check_extrapolation(fd, ext):
-    # Check how to extrapolate
-    # By default uses the value of the fd object
-    if ext == "default" or ext == 0:
-        ext = fd.default_extrapolation
+class ExtrapolationType(enum.Enum):
+    r"""Enum with extrapolation types.
 
-    if ext == "extrapolation" or ext == 1:
-        periodic = 1
-        extrapolation = False
+        Defines the extrapolation mode for elements outside the domain range.
 
-    elif ext == "periodic" or ext == 2:
-        extrapolation = 2
-        periodic = True
+        * default: uses the default method defined in the object to extrapolate.
+        * extrapolation: uses the extrapolated values directly.
+        * periodic: extends the domain range periodically.
+        * const: uses the boundary value.
+        * slice: avoids extrapolation restricting the domain.
+    """
+    default = 0
```

Comment:
The values can be the strings. Do we want the numerical values for **anything**?

Reply:
Changed in a94db9f.